### PR TITLE
Support IPv6 ::1 in listeningOrigin

### PR DESCRIPTION
**Pull request by mcollina**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by ivan-tymoshenko — APPROVED** [review]
lgtm

File: fastify.js [hunk 1]
```diff
@@ -355,7 +355,8 @@ function fastify (options) {
         if (typeof address === 'string') {
           return address
         }
-        return `${this[kOptions].https ? 'https' : 'http'}://${address.address}:${address.port}`
+        const host = address.address === '::1' ? '[::1]' : address.address
```
Comment (climba03003) [inline fastify.js:358]: Would it be better?
```suggestion
        const host = address.family === 'IPv6' ? `[${address.address}]` : address.address
```

**Review by RafaelGSS — APPROVED** [review]

File: fastify.js (hunk 1 above)
Comment (mcollina) [reply fastify.js:358]: I'm not that familiar with URLs and IPv6 to know, will confirm

Comment (jsumners) [reply fastify.js:358]: The suggestion is correct. 

http://[::1]:8000/hello

Comment (jsumners) [reply fastify.js:358]: https://www.rfc-editor.org/rfc/rfc3986#section-1.1.2
https://www.rfc-editor.org/rfc/rfc2732#section-2

**Review by climba03003 — APPROVED** [review]

**Review by jsumners — APPROVED** [review]

**Review by Eomm — APPROVED** [review]

**Pull request merged**